src/SWP/Bundle/AnalyticsBundle/EventListener/KernelListener.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/**
+ * This file is part of the Superdesk Web Publisher Template Engine Bundle.
+ *
+ * Copyright 2015 Sourcefabric z.ú. and contributors.
+ *
+ * For the full copyright and license information, please see the
+ * AUTHORS and LICENSE files distributed with this source code.
+ *
+ * @copyright 2016 Sourcefabric z.ú.
+ * @license http://www.superdesk.org/license
+ */
+namespace SWP\Bundle\AnalyticsBundle\EventListener;
+
+use SWP\Bundle\AnalyticsBundle\Model\RequestMetric;
+use SWP\Bundle\AnalyticsBundle\Repository\RequestMetricRepository;
+use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+use Symfony\Component\Stopwatch\Stopwatch;
+use Symfony\Component\Stopwatch\StopwatchEvent;
+
+class KernelListener
```

Comment:
I assumed we would be adding to the functionality in this bundle. I would
prefer to have just one kernel listener in the AnalyticsBundle. There is
also a chance that there may have to be other classes in the EventListener
folder to handle the metrics, as it's not possible, afaik, to handle
different kinds of events in the same class. So we might have a kernel
listener and a doctrine listener which both handle metrics. So would you
like the class to be renamed MetricsKernelListener, or left as it is until
we have another KernelListener in this bundle?

On Mon, Jun 13, 2016 at 12:36 PM, Paweł Mikołajczuk <
notifications@github.com> wrote:

> In src/SWP/Bundle/AnalyticsBundle/EventListener/KernelListener.php
> <https://github.com/superdesk/web-publisher/pull/78#discussion_r66771323>:
>
> > + * For the full copyright and license information, please see the
> > + * AUTHORS and LICENSE files distributed with this source code.
> > + *
> > + * @copyright 2016 Sourcefabric z.ú.
> > + * @license http://www.superdesk.org/license
> > + */
> > +namespace SWP\Bundle\AnalyticsBundle\EventListener;
> > +
> > +use SWP\Bundle\AnalyticsBundle\Model\RequestMetric;
> > +use SWP\Bundle\AnalyticsBundle\Repository\RequestMetricRepository;
> > +use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
> > +use Symfony\Component\HttpKernel\Event\GetResponseEvent;
> > +use Symfony\Component\Stopwatch\Stopwatch;
> > +use Symfony\Component\Stopwatch\StopwatchEvent;
> > +
> > +class KernelListener
>
> That name is wrong - It should be MetricsListener.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/superdesk/web-publisher/pull/78/files/5fbb2cfc7cad6a7bafa68838427244bcec0eb8fa#r66771323>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ARBtFfN0vZMzylKvapHCAnsn-npE7TFtks5qLTLLgaJpZM4Ix7A->
> .
>